python/tchannel/tornado/stream.py

```diff
@@ -161,8 +161,13 @@ def __init__(self, rpipe, wpipe=None, auto_close=False):
         self.auto_close = auto_close
         self.state = StreamState.init
 
+        self.exception = None
+
     @tornado.gen.coroutine
     def read(self):
+        if self.exception:
+            raise self.exception
```

Comment:
for async exception, what the stack trace we want to keep. We need discuss more about it